### PR TITLE
feat(next): add l10n to localizer for next rsc

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- Ensure all localization strings can be localized in React Server
  Components.

This commit:

- Removes LocalizerServer
- Adds LocalizerRSC to serve as a wrapper for ReactLocalization that can
  be used by React Server Components, as well as a few other utility
  methods.
- Adds LocalizerRscFactory that instantiates LocalizerRSC only with
  bundles for a provided acceptLanguage string.
- Adds LocalizerRscFactoryProvider to be used with the NestApp to
  instantiate and then intialize the LocalizerRscFactory. Initalization
  will fetch all messages from disk and populate the bundles.
- Reorganize shared/l10n library to match Node style guide
- Removes demo l10n code
- Update purchase-details, terms-and-service components, and success and
  error pages to use new LocalizerRSC class for localization.
- Adds temporary l10n-convert script to provide translation for existing
  ftl strings.

## Issue that this pull request solves

Closes: #FXA-8822

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: libs/shared/l10n/src/lib/localizer/localizer.rsc.ts
Comment (pdehaan) [inline libs/shared/l10n/src/lib/localizer/localizer.rsc.ts:1]: Does this need a copyright header?

File: apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/error/page.tsx [hunk 1]
```diff
@@ -73,19 +68,20 @@ export default async function CheckoutError({
       >
         <Image src={errorIcon} alt="" className="mt-16 mb-10" />
         <p className="page-message px-7 py-0 mb-4 ">
-          {l10n
-            .getMessage(getErrorReason(cart.errorReasonId).messageFtl)
-            ?.value?.toString() || getErrorReason(cart.errorReasonId).message}
+          {l10n.getString(
+            getErrorReason(cart.errorReasonId).messageFtl,
+            getErrorReason(cart.errorReasonId).message
```
[the remark this answers is not in the document]
Comment (StaberindeZA) [reply apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/error/page.tsx:73]: Ah yes good idea. I just auto-piloted right through this refactor, and didn't notice. Thank you.

File: libs/shared/l10n/src/lib/l10n.utils.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        .fromNow();
+    },
+  };
+}
+
+/**
+ * Get the best fitting locale, prioritizing request search params, followed by request header AcceptLanguage and DEFAULT_LOCALE as default
+ * @param params - parameters of the request
+ * @param acceptLanguage - Accept language from request header
+ * @returns The best fitting locale
```
Comment (pdehaan) [inline libs/shared/l10n/src/lib/l10n.utils.ts:121]: nit: Missing JSDoc for `supportedLanguages` attribute.

File: libs/shared/l10n/src/lib/localizer/localizer.rsc.factory.spec.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Test, TestingModule } from '@nestjs/testing';
+import { LocalizerRscFactory } from './localizer.rsc.factory';
+import { ILocalizerBindings } from './localizer.interfaces';
+import supportedLanguages from '../supported-languages.json';
+import { LocalizerRSC } from './localizer.rsc';
```
Comment (pdehaan) [inline libs/shared/l10n/src/lib/localizer/localizer.rsc.factory.spec.ts:9]: nit: why `LocalizerRSC` (with uppercases) but then `LocalizerRscFactory` (with lowercases)?

File: libs/shared/l10n/src/lib/localizer/localizer.rsc.factory.spec.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  describe('fetchFluentBundles', () => {
+    it('should succeed', async () => {
+      expect(localizer['bundles'].size).toBe(supportedLanguages.length);
+      expect(localizer['document']).toBeDefined();
+    });
+  });
+
+  describe('createLocalizerRSC', () => {
+    it('should return instance of LocalizerRSC', async () => {
+      const localizerRsc = await localizer.createLocalizerRSC('en');
```
Comment (pdehaan) [inline libs/shared/l10n/src/lib/localizer/localizer.rsc.factory.spec.ts:58]: Related to previous naming nit about sometimes all-caps "RSC" (vs something like `LocalizerRscFactory`).

**Review by xlisachan — APPROVED** [review]
r+! I pulled down the branch and was able to see the page translated when switching language preferences. Nice cleanup and thorough tests, thanks Reino!

File: libs/shared/l10n/src/lib/localizer/localizer.rsc.factory.spec.ts (hunk 3 above)
Comment (StaberindeZA) [reply libs/shared/l10n/src/lib/localizer/localizer.rsc.factory.spec.ts:9]: I asked myself that same thing. Ended up sticking with `*Rsc*`. TY :)

**Review by bcolsson — APPROVED** [review]
Approving FTL files. After discussing with Reino about the purpose/use of the `convert.ts` file as a QA assist and all final strings will eventually be added in future PRs, I don't have any concerns with the patch.

**Pull request merged**